src/client/visualizations/totals/totals.tsx

```diff
@@ -53,14 +57,24 @@ export class Totals extends BaseVisualization<BaseVisualizationState> {
   }
 
   makeQuery(essence: Essence, timekeeper: Timekeeper): Expression {
-    let query: Expression = ply()
-      .apply("main", $("main").filter(essence.getEffectiveFilter(timekeeper, Totals.id).toExpression()));
+    const combineWithPrevious = essence.hasComparison();
 
-    essence.getEffectiveMeasures().forEach(measure => {
-      query = query.performAction(measure.toApplyExpression());
-    });
+    const mainExp: Expression = ply()
+      .apply("main", $("main").filter(essence.getEffectiveFilter(timekeeper, { combineWithPrevious, highlightId: Totals.id }).toExpression()));
+
+    const currentFilter = essence.currentTimeFilter(timekeeper);
+    const previousFilter = combineWithPrevious ? essence.previousTimeFilter(timekeeper) : null;
 
-    return query;
+    return essence.getEffectiveMeasures().reduce((query, measure) => {
```

Comment:
It would be nice to extract duplicated code from different visualisations if feasible.